File: src/main/java/kr/co/bookand/backend/account/domain/SuspendedAccount.java
package kr.co.bookand.backend.account.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
public class SuspendedAccount {

    @Id
    @GeneratedValue
    @Column(name = "revoke_account_id")
    private Long id;

    private int suspendedCount;
    private LocalDateTime startedSuspendedDate;
    private LocalDateTime endedSuspendedDate;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "account_id")
    private Account account;

    public void addSuspendedCount() {
        this.suspendedCount++;
    }

    public void setSuspendedDate(LocalDateTime suspendedDate) {
        this.startedSuspendedDate = LocalDateTime.now();
        this.endedSuspendedDate = suspendedDate;
    }

}

File: src/main/kotlin/kr/co/bookand/backend/account/domain/SuspendedAccount.kt
package kr.co.bookand.backend.account.domain

import java.time.LocalDateTime
import javax.persistence.*

@Entity
class SuspendedAccount(
    @Id
    @GeneratedValue
    @Column(name = "suspended_account_id")
    var id: Long? = 0,

    var suspendedCount: Int = 0,
    var startedSuspendedAt: LocalDateTime? = null,
    var endedSuspendedAt: LocalDateTime? = null,

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    var account: Account
) {

    fun addSuspendedCount() {
        suspendedCount++
    }

    fun setSuspendedAt(suspendedDate: LocalDateTime) {
        this.startedSuspendedAt = LocalDateTime.now()
        this.endedSuspendedAt = suspendedDate
    }
}